datalake/src/main/java/com/sequenceiq/datalake/configuration/CDPConfigService.java

```diff
@@ -130,22 +141,39 @@ public StackV4Request getConfigForKey(CDPConfigKey cdpConfigKey) {
     }
 
     public String getDefaultRuntime() {
-        return defaultRuntime;
+        if (Strings.isNullOrEmpty(defaultRuntime)) {
+            return getDatalakeVersions(null).stream().findFirst()
+                    .orElseThrow(() -> new CloudbreakServiceException("Runtime not found in the default image catalog"));
+        } else {
+            return defaultRuntime;
+        }
     }
 
     public List<String> getDatalakeVersions(String cloudPlatform) {
-        List<String> runtimeVersions = cdpStackRequests.keySet().stream()
+        List<String> runtimeVersions = getRuntimeVersions(cloudPlatform);
+        if (Strings.isNullOrEmpty(defaultRuntime)) {
+            List<String> defaultImageCatalogRuntimeVersions = imageCatalogService.getDefaultImageCatalogRuntimeVersions(DEFAULT_WORKSPACE_ID);
+            LOGGER.debug("Generate runtime versions by checking available runtimes in the default image catalog ('%s').",
```

Comment:
The logger formatting is wrong since this is not `String.format`.

Please change it to:
`Generate runtime versions by checking available runtimes in the default image catalog ('{}')."`

Reply:
:+1: